src/miniscript/types/extra_props.rs

```diff
@@ -659,10 +659,13 @@ impl Property for ExtData {
                 r.max_sat_size.and_then(|(w, s)| Some((w + 1, s + 1))),
             ),
             max_dissat_size: match (l.max_dissat_size, r.max_dissat_size) {
+                (Some(l), Some(r)) => {
+                    let max = cmp::max(l, r);
+                    Some((1 + max.0, 1 + max.1))
+                }
                 (None, Some(r)) => Some((1 + r.0, 1 + r.1)),
-                (Some(l), None) => Some((2 + l.0, 1 + l.1)),
+                (Some(l), None) => Some((1 + l.0, 1 + l.1)),
```

Comment:
Should this not be 2?

Reply:
Yep, as you pass on the first branch of the IF to dissatisfy (?) (same as your comment wrt to `cast_likely` above).